src/cargo/core/compiler/custom_build.rs

```diff
@@ -51,7 +51,7 @@ pub struct BuildOutput {
 /// `Unit` because this structure needs to be shareable between threads.
 #[derive(Default)]
 pub struct BuildScriptOutputs {
-    outputs: HashMap<(PackageId, Metadata), BuildOutput>,
+    outputs: HashMap<Metadata, BuildOutput>,
```

Comment:
I changed this to just be keyed off the Metadata to simplify things.  I don't remember why I originally included `PackageId`, as the PackageId hash is already included in the `Metadata`.  If we are uncomfortable with the possibility of a hash collision, I can switch it back, or we can consider using a stronger hash (like siphash 128?).